obspy/io/sac/util.py

```diff
@@ -159,19 +159,24 @@ def is_same_byteorder(bo1, bo2):
     return (bo1.lower() in le) == (bo2.lower() in le)
 
 
-def _clean_str(value):
+# This fn is used in two places: in SACTrace.read, to sanitize strings for
+# SACTrace, and in sac_to_obspy_header, to sanitize strings for Trace that the
+# user may have manually added.
+def _clean_str(value, strip_whitespace=True):
     # Remove null values and whitespace, return a str
     try:
         # value is a str
         null_term = value.find('\x00')
     except TypeError:
         # value is a bytes
-        # null_term = value.decode().find('\x00')
         null_term = value.find(b'\x00')
+    except UnicodeError:
+        null_term = value.decode('ASCII', 'replace').find(b'\x00')
 
     if null_term >= 0:
-        value = value[:null_term]
-    value = value.strip()
+        value = value[:null_term] + b" " * len(value[null_term:])
```

Comment:
> Can't we just cut the thing off at the termination character (like was done before in the old obspy.sac implementation)?

Except for `kevnm`, SAC string fields are 8 characters long, so I think a termination character before the last position should expect to be whitespaced-over.  In old `obspy.sac` [implementation](https://github.com/obspy/obspy/blob/0.10.2/obspy/sac/sacio.py#L1209), this operation was only used while building a dictionary for `Stats`, which doesn't care for extra whitespace.  `_clean_str` is now being used to also sanitize the NumPy array that contains the raw strings, which does care about trailing whitespace, hence the new flag.

> Also, if value was a string before, you're now adding bytes to it..

I'm still trying to wrap my head around strings vs bytes.  Can you suggest an example where this would fail, and we can add it to the tests?